app/updatemanager.go

```diff
@@ -387,22 +387,41 @@ func (c *ControlMapPool) Get(ID string) (active []*UpdateControlMap, expired []*
 	return active, expired
 }
 
+func (c *ControlMapPool) Delete(ID string) {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+	filteredPool := []*UpdateControlMap{}
+	query(
+		c.Pool,
+		// Collector
+		func(u *UpdateControlMap) {
+			filteredPool = append(filteredPool, u)
+		},
+		// Predicates
+		func(u *UpdateControlMap) bool {
+			return u.ID != ID
+		},
+	)
+	c.Pool = filteredPool
+	c.saveToStore()
+}
```

Comment:
FYI, I "imported" this function to #783 with an extra call to `announceUpdate`, which was missed here.